### ISSUE TITLE
Support for Metadata update from source archive

### ISSUE THREAD

**Issue by y00437830** [artipie/npm-adapter#27]
In addition to interacting with the NPM client, the artifactory repo also support upload NPM .tgz file manually by upload request.
It mean that all the information we can give you is the  .tgz filestream. If you Unzip the .tgz file， you could get the project package.json
Then, you could get the metadata information from the package.json.
I give the detail describtion by email

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) please, pay attention to this issue

**Comment by 0crat** [comment]
@y00437830[/z](https://www.0crat.com/u/y00437830) this project will fix the problem faster if you donate a few dollars to it; just [click here](https://www.0crat.com/contrib/CT2E6TK9B) and pay via Stripe, it's very fast, convenient and appreciated; thanks a lot!

**Comment by Sammers21** [comment]
@y00437830 is referring to a feture, alredy implemented in Artifactory: `'Deploying with cURL'` at https://www.jfrog.com/confluence/display/BT/npm+Repositories

**Comment by Sammers21** [comment]
Hey @g4s8 , @y00437830  wants this feature to be high priority.

**Comment by Sammers21** [comment]
@g4s8, rename the issue to 'Support for Metadata update from source archive'

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 11@e6cb530cd606, thread: PQ-CT2E6TK9B
com.jcabi.xml.StrictXML[124] java.lang.IllegalArgumentException: 2 error(s) in XML document: -1:-1: cvc-pattern-valid: Value 'gh:Adapter back end integration' is not facet-valid with respect to pattern '[a-z]{2}:[A-Z0-9a-z.\-#/]+' for type 'milestoneID'.;-1:-1: cvc-complex-type.2.2: Element 'successor' must have no element [children], and the value must be valid.
```

[0.51.7](https://github.com/zerocracy/farm): CID: [56f73868-1879-4ed5-8c29-48d8d4f77c9a](https://www.0crat.com/footprint/CT2E6TK9B/56f73868-1879-4ed5-8c29-48d8d4f77c9a), Type: "Job milestoned"

**Comment by 0crat** [comment]
The job [#27](https://github.com/artipie/npm-adapter/issues/27) assigned to @sammers21[/z](https://www.0crat.com/u/sammers21), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/d06fbe34-b415-48dc-94bf-e8bc7d77f943); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); @sammers21[/z](https://www.0crat.com/u/sammers21) is not a member of this project yet, but they can request to join, as [§1](http://www.zerocracy.com/policy.html#1) explains; we should be aware that @sammers21[/z](https://www.0crat.com/u/sammers21) is on vacation; this ticket may be delayed; there will be no monetary reward for this job

**Comment by 0crat** [comment]
Thanks for your contribution, @y00437830[/z](https://www.0crat.com/u/y00437830)! If you would be a member of [the project](https://www.0crat.com/p/CT2E6TK9B), you would now earn +15 reputation points, as explained in [§29](http://www.zerocracy.com/policy.html#29). You can join and apply to it, see [§2](http://www.zerocracy.com/policy.html#2).

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by Sammers21** [comment]
@g4s8 , @y00437830 , we are not the Artifactory development office and this feature is absolutely not-mandatory for the module, that is why the priority for the issues should be the lowest possible.

**Comment by y00437830** [comment]
@Sammers21 ,
but We are Huawei employees, we are mainly responsible for the company's business success.  And this situation is the first situation we meet in the company project。 so,  Try our best.

**Comment by Sammers21** [comment]
@y00437830, yes, this is very important. We are planning to implement this feature but on the HTTP layer. Can this feature be implemented on your project-level just for now?

**Comment by y00437830** [comment]
> @y00437830, yes, this is very important. We are planning to implement this feature but on the HTTP layer. Can this feature be implemented on your project-level just for now?

No problem,   if you can not give an enhancement version today or tomorrow.

We are planning to deploy it in this weekend. So, we can wait for two days.  
if the enhancement is not ok.  We will start implementing it on our project-level on Friday.  Since we don't need to think globally for all languages like Adapter & Asto.  so,  the temporary code is not too late for us to start on Friday.
So, don't feel any delivery pressure.

**Comment by Sammers21** [comment]
@y00437830 , Yes, that is right. Implement it on your project yourself

**Comment by g4s8** [comment]
@y00437830 @Sammers21 the problem seems to be solved, can we close this ticket?

**Comment by 0crat** [comment]
The user @sammers21[/z](https://www.0crat.com/u/sammers21) resigned from [#27](https://github.com/artipie/npm-adapter/issues/27), please stop working. Reason for job resignation: It is older than 10 days, see [§8](http://www.zerocracy.com/policy.html#8)

**Comment by 0crat** [comment]
The job [#27](https://github.com/artipie/npm-adapter/issues/27) assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/f2059492-0cc8-4243-be45-2ebf810e06d4); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) this job was assigned to you 6days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0pdd** [comment]
@y00437830 the puzzle [#36](https://github.com/artipie/npm-adapter/issues/36) is still not solved.

**Comment by paulodamaso** [comment]
@g4s8 Please close,https://github.com/artipie/npm-adapter/pull/35 has already been merged

**Comment by Sammers21** [comment]
@paulodamaso , why do you ask to close if the issue is not resolved?

**Comment by paulodamaso** [comment]
@Sammers21 We use PDD to propagate issue resolution, which means that means that may generate a thread of puzzles to be solved. I got paid by each issue I close, which is equivalent to 30 minutes of work. This issue is closed so I can receive my payment for the deliverable I've produced related to this, and I left a puzzle for us to continue to resolve the issue thread incrementally (generally in 30-minute increments).
